[BUG] NoClassDefFoundError: com/nvidia/spark/rapids/shims/PlanShims in UCX tests

**Describe the bug**
UCX nightly 400+ tests failed 
```
[2022-09-08T01:17:45.659Z]                         format(target_id, ".", name), value)
[2022-09-08T01:17:45.659Z] [1m[31mE                   py4j.protocol.Py4JJavaError: An error occurred while calling z:com.nvidia.spark.rapids.ExecutionPlanCaptureCallback.assertDidFallBack.[0m
[2022-09-08T01:17:45.659Z] [1m[31mE                   : java.lang.NoClassDefFoundError: com/nvidia/spark/rapids/shims/PlanShims$[0m
[2022-09-08T01:17:45.659Z] [1m[31mE                   	at com.nvidia.spark.rapids.ExecutionPlanCaptureCallback$.extractExecutedPlan(Plugin.scala:424)[0m
[2022-09-08T01:17:45.659Z] [1m[31mE                   	at com.nvidia.spark.rapids.ExecutionPlanCaptureCallback$.assertDidFallBack(Plugin.scala:437)[0m
[2022-09-08T01:17:45.659Z] [1m[31mE                   	at com.nvidia.spark.rapids.ExecutionPlanCaptureCallback$.assertDidFallBack(Plugin.scala:444)[0m
[2022-09-08T01:17:45.659Z] [1m[31mE                   	at com.nvidia.spark.rapids.ExecutionPlanCaptureCallback.assertDidFallBack(Plugin.scala)[0m
[2022-09-08T01:17:45.659Z] [1m[31mE                   	at sun.reflect.GeneratedMethodAccessor290.invoke(Unknown Source)[0m
[2022-09-08T01:17:45.659Z] [1m[31mE                   	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)[0m
[2022-09-08T01:17:45.659Z] [1m[31mE                   	at java.lang.reflect.Method.invoke(Method.java:498)[0m
[2022-09-08T01:17:45.659Z] [1m[31mE                   	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)[0m
[2022-09-08T01:17:45.659Z] [1m[31mE                   	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)[0m
[2022-09-08T01:17:45.659Z] [1m[31mE                   	at py4j.Gateway.invoke(Gateway.java:282)[0m
[2022-09-08T01:17:45.659Z] [1m[31mE                   	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)[0m
[2022-09-08T01:17:45.659Z] [1m[31mE                   	at py4j.commands.CallCommand.execute(CallCommand.java:79)[0m
[2022-09-08T01:17:45.659Z] [1m[31mE                   	at py4j.GatewayConnection.run(GatewayConnection.java:238)[0m
[2022-09-08T01:17:45.659Z] [1m[31mE                   	at java.lang.Thread.run(Thread.java:750)[0m
```

list of failed tests
```
[2022-09-08T01:17:45.663Z] FAILED ../../src/main/python/arithmetic_ops_test.py::test_multiplication_fallback_when_ansi_enabled[Byte][ALLOW_NON_GPU(ProjectExec,Alias,Multiply,Cast)]
[2022-09-08T01:17:45.663Z] FAILED ../../src/main/python/arithmetic_ops_test.py::test_multiplication_fallback_when_ansi_enabled[Short][ALLOW_NON_GPU(ProjectExec,Alias,Multiply,Cast)]
[2022-09-08T01:17:45.663Z] FAILED ../../src/main/python/arithmetic_ops_test.py::test_multiplication_fallback_when_ansi_enabled[Integer][ALLOW_NON_GPU(ProjectExec,Alias,Multiply,Cast)]
[2022-09-08T01:17:45.663Z] FAILED ../../src/main/python/arithmetic_ops_test.py::test_multiplication_fallback_when_ansi_enabled[Long][ALLOW_NON_GPU(ProjectExec,Alias,Multiply,Cast)]
[2022-09-08T01:17:45.663Z] FAILED ../../src/main/python/arithmetic_ops_test.py::test_pmod_fallback[Decimal(38,0)][ALLOW_NON_GPU(ProjectExec,Pmod)]
[2022-09-08T01:17:45.663Z] FAILED ../../src/main/python/arithmetic_ops_test.py::test_pmod_fallback[Decimal(38,10)][ALLOW_NON_GPU(ProjectExec,Pmod)]
[2022-09-08T01:17:45.663Z] FAILED ../../src/main/python/csv_test.py::test_csv_fallback[spark.rapids.sql.format.csv.enabled-read_csv_df][ALLOW_NON_GPU(org.apache.spark.sql.execution.LeafExecNode)]
[2022-09-08T01:17:45.663Z] FAILED ../../src/main/python/csv_test.py::test_csv_fallback[spark.rapids.sql.format.csv.enabled-read_csv_sql][ALLOW_NON_GPU(org.apache.spark.sql.execution.LeafExecNode)]
...
...
...
[2022-09-08T01:17:45.666Z] FAILED ../../src/main/python/parquet_write_test.py::test_parquet_writeLegacyFormat_fallback[ALLOW_NON_GPU(DataWritingCommandExec)]
[2022-09-08T01:17:45.666Z] FAILED ../../src/main/python/parquet_write_test.py::test_buckets_write_fallback[IGNORE_ORDER, ALLOW_NON_GPU(DataWritingCommandExec)]
[2022-09-08T01:17:45.666Z] FAILED ../../src/main/python/regexp_test.py::test_split_regexp_disabled_fallback[ALLOW_NON_GPU(ProjectExec,StringSplit)]
[2022-09-08T01:17:45.666Z] FAILED ../../src/main/python/regexp_test.py::test_re_replace_issue_5492[ALLOW_NON_GPU(ProjectExec,RegExpReplace)]
[2022-09-08T01:17:45.666Z] FAILED ../../src/main/python/regexp_test.py::test_rlike_fallback_empty_group[ALLOW_NON_GPU(ProjectExec,RLike)]
[2022-09-08T01:17:45.666Z] FAILED ../../src/main/python/regexp_test.py::test_rlike_fallback_possessive_quantifier[ALLOW_NON_GPU(ProjectExec,RLike)]
[2022-09-08T01:17:45.666Z] FAILED ../../src/main/python/regexp_test.py::test_regexp_memory_fallback[ALLOW_NON_GPU(ProjectExec,RLike)]
[2022-09-08T01:17:45.666Z] FAILED ../../src/main/python/repart_test.py::test_round_robin_sort_fallback[[('ag', Array(String))]][IGNORE_ORDER({'local': True}), ALLOW_NON_GPU(ShuffleExchangeExec,RoundRobinPartitioning)]
[2022-09-08T01:17:45.666Z] FAILED ../../src/main/python/repart_test.py::test_round_robin_sort_fallback[[('a', Struct(('a_1', Struct(('a_1_1', Integer),('a_1_2', Float),('a_1_3', Double))),('b_1', Long),('c_1', Integer)))]][IGNORE_ORDER({'local': True}), ALLOW_NON_GPU(ShuffleExchangeExec,RoundRobinPartitioning)]
[2022-09-08T01:17:45.666Z] FAILED ../../src/main/python/repart_test.py::test_round_robin_sort_fallback[[('a', Map(String(not_null),String))]][IGNORE_ORDER({'local': True}), ALLOW_NON_GPU(ShuffleExchangeExec,RoundRobinPartitioning)]
[2022-09-08T01:17:45.666Z] FAILED ../../src/main/python/repart_test.py::test_hash_repartition_exact_fallback[([('ag', Array(Struct(('b1', Long))))], ['ag'])-2][IGNORE_ORDER({'local': True}), ALLOW_NON_GPU(ProjectExec,ShuffleExchangeExec)]
[2022-09-08T01:17:45.666Z] FAILED ../../src/main/python/repart_test.py::test_hash_repartition_exact_fallback[([('ag', Array(Struct(('b1', Long))))], ['ag'])-10][IGNORE_ORDER({'local': True}), ALLOW_NON_GPU(ProjectExec,ShuffleExchangeExec)]
[2022-09-08T01:17:45.666Z] FAILED ../../src/main/python/repart_test.py::test_hash_repartition_exact_fallback[([('ag', Array(Struct(('b1', Long))))], ['ag'])-17][IGNORE_ORDER({'local': True}), ALLOW_NON_GPU(ProjectExec,ShuffleExchangeExec)]
[2022-09-08T01:17:45.666Z] FAILED ../../src/main/python/repart_test.py::test_hash_repartition_exact_fallback[([('ag', Array(Struct(('b1', Long))))], ['ag'])-19][IGNORE_ORDER({'local': True}), ALLOW_NON_GPU(ProjectExec,ShuffleExchangeExec)]
[2022-09-08T01:17:45.666Z] FAILED ../../src/main/python/repart_test.py::test_hash_repartition_exact_fallback[([('ag', Array(Struct(('b1', Long))))], ['ag'])-32][IGNORE_ORDER({'local': True}), ALLOW_NON_GPU(ProjectExec,ShuffleExchangeExec)]
[2022-09-08T01:17:45.666Z] FAILED ../../src/main/python/sort_test.py::test_sort_binary_fallback[Column<'CAST(a AS BINARY)'>-String(not_null)][ALLOW_NON_GPU(SortExec,ShuffleExchangeExec,RangePartitioning,SortOrder)]
[2022-09-08T01:17:45.666Z] FAILED ../../src/main/python/sort_test.py::test_single_nested_orderby_fallback_for_nullorder[Column<'a ASC NULLS LAST'>-Struct(['child0', Byte],['child1', Short],['child2', Integer],['child3', Long],['child4', Float],['child5', Double],['child6', String],['child7', Boolean],['child8', Date],['child9', Timestamp],['child10', Null])][ALLOW_NON_GPU(SortExec,ShuffleExchangeExec,RangePartitioning,SortOrder)]
[2022-09-08T01:17:45.666Z] FAILED ../../src/main/python/sort_test.py::test_single_nested_orderby_fallback_for_nullorder[Column<'a ASC NULLS LAST'>-Struct(['child0', Struct(['child0', Byte],['child1', Short],['child2', Integer],['child3', Long],['child4', Float],['child5', Double],['child6', String],['child7', Boolean],['child8', Date],['child9', Timestamp],['child10', Null])])][ALLOW_NON_GPU(SortExec,ShuffleExchangeExec,RangePartitioning,SortOrder)]
[2022-09-08T01:17:45.666Z] FAILED ../../src/main/python/sort_test.py::test_single_nested_orderby_fallback_for_nullorder[Column<'a DESC NULLS FIRST'>-Struct(['child0', Byte],['child1', Short],['child2', Integer],['child3', Long],['child4', Float],['child5', Double],['child6', String],['child7', Boolean],['child8', Date],['child9', Timestamp],['child10', Null])][ALLOW_NON_GPU(SortExec,ShuffleExchangeExec,RangePartitioning,SortOrder)]
[2022-09-08T01:17:45.666Z] FAILED ../../src/main/python/sort_test.py::test_single_nested_orderby_fallback_for_nullorder[Column<'a DESC NULLS FIRST'>-Struct(['child0', Struct(['child0', Byte],['child1', Short],['child2', Integer],['child3', Long],['child4', Float],['child5', Double],['child6', String],['child7', Boolean],['child8', Date],['child9', Timestamp],['child10', Null])])][ALLOW_NON_GPU(SortExec,ShuffleExchangeExec,RangePartitioning,SortOrder)]
[2022-09-08T01:17:45.666Z] FAILED ../../src/main/python/sort_test.py::test_single_nested_orderby_with_limit_fallback[Column<'a ASC NULLS LAST'>-Struct(['child0', Byte],['child1', Short],['child2', Integer],['child3', Long],['child4', Float],['child5', Double],['child6', String],['child7', Boolean],['child8', Date],['child9', Timestamp],['child10', Null])][ALLOW_NON_GPU(TakeOrderedAndProjectExec,SortOrder)]
[2022-09-08T01:17:45.666Z] FAILED ../../src/main/python/sort_test.py::test_single_nested_orderby_with_limit_fallback[Column<'a DESC NULLS FIRST'>-Struct(['child0', Byte],['child1', Short],['child2', Integer],['child3', Long],['child4', Float],['child5', Double],['child6', String],['child7', Boolean],['child8', Date],['child9', Timestamp],['child10', Null])][ALLOW_NON_GPU(TakeOrderedAndProjectExec,SortOrder)]
[2022-09-08T01:17:45.666Z] FAILED ../../src/main/python/sort_test.py::test_single_nested_orderby_with_limit_fallback[Column<'a ASC NULLS FIRST'>-Array(String)][ALLOW_NON_GPU(TakeOrderedAndProjectExec,SortOrder)]
[2022-09-08T01:17:45.666Z] FAILED ../../src/main/python/sort_test.py::test_single_nested_orderby_with_limit_fallback[Column<'a ASC NULLS LAST'>-Array(String)][ALLOW_NON_GPU(TakeOrderedAndProjectExec,SortOrder)]
[2022-09-08T01:17:45.666Z] FAILED ../../src/main/python/sort_test.py::test_single_nested_orderby_with_limit_fallback[Column<'a DESC NULLS LAST'>-Array(String)][ALLOW_NON_GPU(TakeOrderedAndProjectExec,SortOrder)]
[2022-09-08T01:17:45.666Z] FAILED ../../src/main/python/sort_test.py::test_single_nested_orderby_with_limit_fallback[Column<'a DESC NULLS FIRST'>-Array(String)][ALLOW_NON_GPU(TakeOrderedAndProjectExec,SortOrder)]
[2022-09-08T01:17:45.666Z] FAILED ../../src/main/python/string_test.py::test_concat_ws_sql_col_sep_only_sep_specified[ALLOW_NON_GPU(ProjectExec,Alias,ConcatWs)]
[2022-09-08T01:17:45.666Z] FAILED ../../src/main/python/window_function_test.py::test_nested_part_fallback[partBy:Array(Long)][IGNORE_ORDER({'local': True}), ALLOW_NON_GPU(WindowExec,Alias,WindowExpression,AggregateExpression,Count,WindowSpecDefinition,SpecifiedWindowFrame,Literal,SortExec,SortOrder,ShuffleExchangeExec,HashPartitioning)]
[2022-09-08T01:17:45.666Z] FAILED ../../src/main/python/window_function_test.py::test_nested_part_fallback[partBy:Struct(['a', Struct(['a1', Long])])][IGNORE_ORDER({'local': True}), ALLOW_NON_GPU(WindowExec,Alias,WindowExpression,AggregateExpression,Count,WindowSpecDefinition,SpecifiedWindowFrame,Literal,SortExec,SortOrder,ShuffleExchangeExec,HashPartitioning)]
```

**Steps/Code to reproduce bug**
Please provide a list of steps or a code sample to reproduce the issue.
Avoid posting private or sensitive data.

**Expected behavior**
A clear and concise description of what you expected to happen.

**Environment details (please complete the following information)**
 - Environment location: [Standalone, YARN, Kubernetes, Cloud(specify cloud provider)]
 - Spark configuration settings related to the issue

**Additional context**
Add any other context about the problem here.

--- a/sql-plugin/src/main/311until320-all/scala/com/nvidia/spark/rapids/shims/PlanShimsImpl.scala
+++ b/sql-plugin/src/main/311until320-all/scala/com/nvidia/spark/rapids/shims/PlanShimsImpl.scala
@@ -16,11 +16,10 @@
 
 package com.nvidia.spark.rapids.shims
 
-import org.apache.spark.sql.execution.{CommandResultExec, SparkPlan}
+import com.nvidia.spark.rapids.PlanShims
 
-object PlanShims {
-  def extractExecutedPlan(plan: SparkPlan): SparkPlan = plan match {
-    case p: CommandResultExec => p.commandPhysicalPlan
-    case _ => plan
-  }
+import org.apache.spark.sql.execution.SparkPlan
+
+class PlanShimsImpl extends PlanShims {
+  def extractExecutedPlan(plan: SparkPlan): SparkPlan = plan
 }

--- a/sql-plugin/src/main/320+/scala/com/nvidia/spark/rapids/shims/PlanShimsImpl.scala
+++ b/sql-plugin/src/main/320+/scala/com/nvidia/spark/rapids/shims/PlanShimsImpl.scala
@@ -16,8 +16,13 @@
 
 package com.nvidia.spark.rapids.shims
 
-import org.apache.spark.sql.execution.SparkPlan
+import com.nvidia.spark.rapids.PlanShims
 
-object PlanShims {
-  def extractExecutedPlan(plan: SparkPlan): SparkPlan = plan
+import org.apache.spark.sql.execution.{CommandResultExec, SparkPlan}
+
+class PlanShimsImpl extends PlanShims {
+  def extractExecutedPlan(plan: SparkPlan): SparkPlan = plan match {
+    case p: CommandResultExec => p.commandPhysicalPlan
+    case _ => plan
+  }
 }

--- a/sql-plugin/src/main/scala/com/nvidia/spark/rapids/PlanShims.scala
+++ b/sql-plugin/src/main/scala/com/nvidia/spark/rapids/PlanShims.scala
@@ -1,0 +1,31 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids
+
+import org.apache.spark.sql.execution.SparkPlan
+
+trait PlanShims {
+  def extractExecutedPlan(plan: SparkPlan): SparkPlan
+}
+
+object PlanShims {
+  private lazy val shims = ShimLoader.newPlanShims()
+
+  def extractExecutedPlan(plan: SparkPlan): SparkPlan = {
+    shims.extractExecutedPlan(plan)
+  }
+}

--- a/sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala
+++ b/sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala
@@ -27,7 +27,6 @@ import scala.util.matching.Regex
 
 import ai.rapids.cudf.{CudaException, CudaFatalException, CudfException, MemoryCleaner}
 import com.nvidia.spark.rapids.python.PythonWorkerSemaphore
-import com.nvidia.spark.rapids.shims.PlanShims
 
 import org.apache.spark.{ExceptionFailure, SparkConf, SparkContext, TaskFailedReason}
 import org.apache.spark.api.plugin.{DriverPlugin, ExecutorPlugin, PluginContext}

--- a/sql-plugin/src/main/scala/com/nvidia/spark/rapids/ShimLoader.scala
+++ b/sql-plugin/src/main/scala/com/nvidia/spark/rapids/ShimLoader.scala
@@ -409,4 +409,8 @@ object ShimLoader extends Logging {
 
   def newIcebergProvider(): IcebergProvider = ShimLoader.newInstanceOf[IcebergProvider](
     "com.nvidia.spark.rapids.iceberg.IcebergProviderImpl")
+
+  def newPlanShims(): PlanShims = ShimLoader.newInstanceOf[PlanShims](
+    "com.nvidia.spark.rapids.shims.PlanShimsImpl"
+  )
 }

